tests/test_parsl_flux_mpi_hello.py

```diff
@@ -85,6 +85,17 @@ def test_flux_resource_list(load_config):
                       stderr=('parsl_flux_resource_list.err', 'w'),
                       env=load_config["environment"]).result()
     assert r == 0
+
+    pattern = re.compile(
+      r'(\s+)STATE(\s+)NNODES(\s+)NCORES(\s+)NGPUS(\s+)NODELIST\n'
+      r'(\s+)free(\s+)(\d+)(\s+)(\d+)(\s+)(\d+)(\s+)(\S+)\n'
+      r'(\s+)allocated(\s+)(\d+)(\s+)(\d+)(\s+)(\d+)(\s+)(\S+)\n'
+      r'(\s+)down(\s+)(\d+)(\s+)(\d+)(\s+)(\d+)'
+      r'.*\n',
+    re.DOTALL)
```

Comment:
I think it's almost there.  The only comment I have is that it could match output that changes in the invariant parts.  For example, the following incorrect output would pass the test:
```
     STATE NNODES   NCORES    NGPUS NODELIST
      free    100       47        8 slurmnode[1-3]
 allocated    999        9        8 slurmnode3
      down      9        9        9 
```

I think it would be pretty easy to tweak your pattern to make sure the invariant parts of the pattern are matched exactly.

Reply:
Oh! Does the pattern above have all the numbers match except the `free` state nodes?
For example, the rows with `allocated` and `down` will always be what they are in the example below? I can also update the entry `slurmnode[1-3]` if it will always be on the form `<host>[\d-\d]`
```
     STATE NNODES   NCORES    NGPUS NODELIST
      free      3       47        0 slurmnode[1-3]
 allocated      1        1        0 slurmnode3
      down      0        0        0 
```

Reply:
Also just checking that the `free` row and `NGPUS` column will always have a value of `0` as well.

Reply:
I see now that when you mean the free nodes will vary, you only mean the `NNODES`. I'll update that.

Reply:
The only thing that should be allowed to change is the number of free cores (under `NCORES`) and the names of the hosts under `NODELIST`.  The number of free nodes, the number of allocated cores and nodes, the number of down nodes and cores should never change.  So, the `47` can change to other numbers, and the names of the hosts can change.  That's it.